storage/src/main/java/org/apache/uniffle/storage/common/LocalStorage.java

```diff
@@ -78,9 +78,15 @@ private LocalStorage(Builder builder) {
 
     File baseFolder = new File(basePath);
     try {
-      if (isEmptyAndWritableDir(baseFolder)) {
-        LOG.warn("Base directory is already an empty dir, skip init");
+      // base folder could be the mounted path, which could not be deleted.
+      // just clean the directory instead.
+      if (baseFolder.isDirectory()) {
+        LOG.info("Base directory is already existed, just clean the directory");
+        FileUtils.cleanDirectory(baseFolder);
       } else {
+        // base folder could be not existed or a regular file instead of dir.
+        // just try to deleteDirectory and recreate it. `deleteDirectory` would throw an exception
+        // when the base folder is a file.
         FileUtils.deleteDirectory(baseFolder);
```

Comment:
Is it still necessary? Because we already know `baseFolder.isDirectory() == false`.

Although one possible case is `baseFolder` being a symlink to a directory.